src/views/InvolvementProfile/components/Membership/index.jsx

```diff
@@ -33,7 +35,11 @@ const Membership = ({ isAdmin, isSiteAdmin, involvementDescription, toggleIsAdmi
         setFollowersNum(followersNum);
         setMembersNum(membersNum);
 
-        if ((membership && membership.Participation !== Participation.Guest) || isSiteAdmin) {
+        setIsMemberAndNonGuest(
+          (membership && membership.Participation !== Participation.Guest) || isSiteAdmin,
+        );
+
+        if (isMemberAndNonGuest) {
```

Comment:
This will not work (reliably) because `setState` does not run sequentially, the state update gets queued to run in the future. So the `setIsMemberAndNonGuest` call will schedule the update to `isMemberAndNonGuest` but within this function the value of `isMemberAndNonGuest` will not immediately change so the `if` on line 42 won't be true when it ought to be.